pkg/storage/fcNode.go

```diff
@@ -276,3 +283,69 @@ func (fc *fcStorage) NodeGetCapabilities(ctx context.Context, req *csi.NodeGetCa
 func (fc *fcStorage) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
 	return nil, status.Error(codes.Unimplemented, "NodeGetInfo is not implemented")
 }
+
+// Retrieve FC initiators for use by controller code for publishing volumes
+func GetFCInitiators() ([]string, error) {
+	specifiedFCAddrs, err := readFCAddrFile()
+	if err != nil {
+		klog.Warningf("Error reading fc address config file: %v", err)
+	}
+	if specifiedFCAddrs != nil {
+		return specifiedFCAddrs, nil
+	}
+
+	klog.Infof("begin FC address discovery")
+	fcAddrFilename := "port_name"
+	scsiHostBasePath := "/sys/class/fc_host/"
+
+	dirList, err := os.ReadDir(scsiHostBasePath)
+	if err != nil {
+		return nil, err
+	}
+
+	discoveredFCAddresses := []string{}
+	for _, hostDir := range dirList {
+		fcAddrFile := filepath.Join(scsiHostBasePath, hostDir.Name(), fcAddrFilename)
+		addrBytes, err := os.ReadFile(fcAddrFile)
+		address := string(addrBytes)
+		address = strings.TrimLeft(strings.TrimRight(address, "\n"), "0x")
+
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			} else {
+				klog.Warningf("error searching for FC HBA addresses: %v", err)
+				return nil, err
+			}
+		} else {
+			klog.Infof("found FC initiator address %s", address)
+			discoveredFCAddresses = append(discoveredFCAddresses, address)
+		}
+	}
+	return discoveredFCAddresses, nil
+}
+
+// Read the fc address configuration file and return addresses
+func readFCAddrFile() ([]string, error) {
```

Comment:
Suggest passing in filename string instead of hard coding it here.

Reply:
Passing filename as suggested in updated version